src/server/sample_data/routes/install.js

```diff
@@ -49,87 +49,95 @@ export const createInstallRoute = () => ({
-      } catch (err) {
-        // ignore delete errors
-      }
-
-      try {
-        const createIndexParams = {
-          index: index,
-          body: {
-            settings: {
-              index: {
-                number_of_shards: 1,
-                number_of_replicas: 0
-              }
-            },
-            mappings: {
-              _doc: {
-                properties: sampleDataset.fields
-              }
-            }
-          }
-        };
-        await callWithRequest(request, 'indices.create', createIndexParams);
-      } catch (err) {
-        const errMsg = `Unable to create sample data index "${index}", error: ${err.message}`;
-        server.log(['warning'], errMsg);
-        return reply(errMsg).code(err.status);
-      }
 
       const now = request.query.now ? request.query.now : new Date();
       const nowReference = dateToIso8601IgnoringTime(now);
-      function updateTimestamps(doc) {
-        sampleDataset.timeFields.forEach(timeFieldName => {
+      function updateTimestamps(doc, dataIndexConfig) {
+        dataIndexConfig.timeFields.forEach(timeFieldName => {
           if (doc[timeFieldName]) {
-            doc[timeFieldName] = sampleDataset.preserveDayOfWeekTimeOfDay
-              ? translateTimeRelativeToWeek(doc[timeFieldName], sampleDataset.currentTimeMarker, nowReference)
-              : translateTimeRelativeToDifference(doc[timeFieldName], sampleDataset.currentTimeMarker, nowReference);
+            doc[timeFieldName] = dataIndexConfig.preserveDayOfWeekTimeOfDay
+              ? translateTimeRelativeToWeek(doc[timeFieldName], dataIndexConfig.currentTimeMarker, nowReference)
+              : translateTimeRelativeToDifference(doc[timeFieldName], dataIndexConfig.currentTimeMarker, nowReference);
           }
         });
         return doc;
       }
-      const bulkInsert = async (docs) => {
-        const bulk = [];
-        docs.forEach(doc => {
-          bulk.push(insertCmd);
-          bulk.push(updateTimestamps(doc));
-        });
-        const resp = await callWithRequest(request, 'bulk', { body: bulk });
-        if (resp.errors) {
-          server.log(
-            ['warning'],
-            `sample_data install errors while bulk inserting. Elasticsearch response: ${JSON.stringify(resp, null, ' ')}`);
-          return Promise.reject(new Error(`Unable to load sample data into index "${index}", see kibana logs for details`));
+
+      const counts = {};
+      for (let i = 0; i < sampleDataset.dataIndices.length; i++) {
+        const dataIndexConfig = sampleDataset.dataIndices[i];
+        const index = createIndexName(sampleDataset.id, dataIndexConfig.id);
+        const insertCmd = {
+          index: {
+            _index: index
+          }
+        };
+
+        // clean up any old installation of dataset
+        try {
+          await callWithRequest(request, 'indices.delete', { index: index });
+        } catch (err) {
+          // ignore delete errors
```

Comment:
Might want to warn here? Probably not important.

Reply:
Do not want to warn because it will create a lot of noise. The delete will fail when the index does not exist.